### PR TITLE
Boundary conflict

### PR DESCRIPTION
**Pull request by BarisYazici**
- If robot ends up in a infeasible state after one step. Transport it to closest boundary point.

- For some reason when I set the joint velocities to zero it transports the robot to origin. Is it the normal behaviour? @letrend

### CODE REVIEW

**Comment by letrend** [comment]
Are you sure you are not changing joint position accidentally?

**Comment by tomasruizt** [comment]
@BarisYazici Integration tests fail. Please run them.

**Comment by tomasruizt** [comment]
The problem is related to not setting the vels to 0 :(

**Comment by BarisYazici** [comment]
> Are you sure you are not changing joint position accidentally?

No I was changing the qd and joint_state[i][0]

> Integration tests fail. Please run them.

Ok checking

**Comment by BarisYazici** [comment]
>The problem is related to not setting the vels to 0 :(

The problem with that is when I set the velocities to zero robot is being teleported to origin.

**Comment by BarisYazici** [comment]
I found the problem with causes integration tests to fail. In gym_reset I don't reset the qd since I refactor the resetting function

File: src/robots/msj_platform.cpp
Comment (tomasruizt) [inline src/robots/msj_platform.cpp]: If we are consistent with `robot.cpp` line 649, in this line you are actually setting the `joint_len` to 0 instead of the `joint_vel`.

File: src/robots/msj_platform.cpp
Comment (tomasruizt) [inline src/robots/msj_platform.cpp]: do you need to define distance in advance, or can you define it where you set it?

File: src/robots/msj_platform.cpp
Comment (tomasruizt) [inline src/robots/msj_platform.cpp]: Should we be using here q3 instead of accessing q[2] ??

File: src/robots/msj_platform.cpp
Comment (tomasruizt) [inline src/robots/msj_platform.cpp:211]: Question: Why do we first update, if we have not yet set the new commanded tendon velocity?

**Review by tomasruizt — CHANGES_REQUESTED** [review]

File: src/robots/msj_platform.cpp
Comment (BarisYazici) [reply src/robots/msj_platform.cpp]: True. Fixed that

File: src/robots/msj_platform.cpp
Comment (BarisYazici) [reply src/robots/msj_platform.cpp:211]: I don't remember that, it's not done in this pr. I could try without but it works now with this.

File: src/robots/msj_platform.cpp
Comment (BarisYazici) [reply src/robots/msj_platform.cpp]: true

File: src/robots/msj_platform.cpp
Comment (BarisYazici) [reply src/robots/msj_platform.cpp]: it can be

**Comment by tomasruizt** [comment]
@BarisYazici approved :+1: I dont have write access to the repo yet, so I cant merge it myself.

File: src/robots/msj_platform.cpp
Comment (tomasruizt) [reply src/robots/msj_platform.cpp:211]: Lets leave it for now

**Pull request merged**
